index.rst

```diff
@@ -48,14 +48,135 @@
+For an integrated butler or butler adjacent implementation, transferring records between registries would be similar to that involved in transferring other records, although the butler adjacent approach would require some thought on how integrated those plugins are with registry.
+
+Open Questions
+==============
+
+* Should we allow commenting on any dimension record?
+  That would be the more general solution although it is only really helpful for ``visit`` and that would normally inherit the properties of ``exposure`` through provenance.
+* Should it be possible to constrain a graph build by either a tag or a quality flag?
+* Should dataset provenance affect the graph build if a parent dataset has a particular quality flag set even though the specific input dataset does not?
+* Summit users require the ability to tag specific error conditions such as vignetting or bad focus.
+  These are annotations against exposure records rather than explicit datasets and is somewhat orthogonal to the limited set of proposed quality flags.
+  Should Butler support tags on dimensions?
+  Should the set of tags be arbitrary, curated, or restricted per telescope?
+* For datasets should quality flagging or applying of tags be handled solely by using TAGGED collections for datasets?
+  For dimensions records a TAGGED collection can not be used (we could though automatically move raw datasets into tagged collections as the flag changes) given that there is a need to be able to ask which exposure records have been flagged regardless of where the datasets are stored.
+* Should we allow commenting on any dimension record?
+  That would be the more general solution although it is only really helpful for ``visit`` and that would normally inherit the properties of ``exposure`` through provenance.
+* How is user information verified?
+  (this relates to the use of butler client/server :cite:`DMTN-176`)
+* Can bots set a quality flag and override a human setting?
+* If multiple people (or bots) disagree on a quality flag should the system keep track of all the changes to the quality flag?
+
+Design Options
+==============
+
+Each dataset in a Butler is identified by a unique dataset ID that is now a UUID.
+For raw data this UUID is predictable and is guaranteed to be the same in all Butler repositories (if ingested into the same run collection).
+A dimension record is also uniquely identifiable (within that record's definition) by an integer.
+
+Core Butler
+-----------
+
+To integrate dataset annotation directly into the Butler registry would be the most efficient solution in terms of performance and ease of distribution to all butler users.
+It would also simplify any requirement to include the quality information in the graph build.
+
+In this scenario the dataset annotations would be stored directly in the Registry database with an associated manager class.
```

Comment:
So you're thinking, roughly:
 - one table for all kinds of annotations, with its own primary key; some of these are unique and some are reusable
 - join tables that relate annotations to datasets and data IDs

?